src/aggregation/aggregator_wrapper.py

```diff
@@ -11,15 +11,15 @@ def __init__(self, clip):
         self.transformer = clip.transformer
         self.ln_final = clip.ln_final
         self.text_projection = clip.text_projection
-        self.attn_mask = clip.attn_mask.cuda()
-        # device = self.token_embedding.device
-        # self.attn_mask = self.attn_mask.cuda()
+        self.attn_mask = clip.attn_mask
 
     def forward(self, text):
+        device = next(self.transformer.parameters()).device
```

Comment:
why do we need this?

Reply:
the clip.attn_mask is not a parameter of the VideoCLIP module and so when the model_video.to(device) is called the attn_mask stays on CPU, so this was just a hack to get the device of the aggregator and manually switch the clip.attn_mask to the correct device